source/Host.h

```diff
@@ -44,25 +125,155 @@ class Host: public Organism {
     }
   }
 
+
+  /**
+   * Input: None
+   * 
+   * Output: None
+   * 
+   * Purpose: To force a copy constructor to be generated by the compiler.
+   */
   Host(const Host &) = default;
+
+
+  /**
+   * Input: None
+   * 
+   * Output: None
+   * 
+   * Purpose: To froce a move constructor to be generated by the compiler
+   */  
   Host(Host &&) = default;
+
+
+  /**
+   * Input: None
+   * 
+   * Output: None
+   * 
+   * Purpose: To tell the compiler to use its default generated variants of the constructor 
+   */
   Host() = default;
 
+
+  /**
+   * Input: None
+   * 
+   * Output: None
+   * 
+   * Purpose: To force a copy assignment operator to be generated by the compiler.
+   */
   Host & operator=(const Host &) = default;
+
+
+  /**
+   * Input: None
+   * 
+   * Output: None
+   * 
+   * Purpose: To force a move assignment operator to be generated by the compiler.
+   */
   Host & operator=(Host &&) = default;
+
+
+  /**
+   * Input: An object of host, and the address of the thing it is being
```

Comment:
This actually just takes another Host, not the address and only one thing since it's passing by reference

Reply:
And compares it to the current host